pyro/burgers/advective_fluxes.py

```diff
@@ -0,0 +1,262 @@
+
+    state.v(buf=1)[:, :] = np.where(l.v(buf=1) > r.v(buf=1), shock.v(buf=1), rarefac.v(buf=1))
+
+    return state
+
+
+def unsplit_fluxes(my_data, rp, dt, scalar_name):
+    r"""
+    Construct the fluxes through the interfaces for the burgers equation:
+
+    .. math::
+
+       u_t  + u u_x  + v u_y  = 0
+       v_t  + u v_x  + v v_y  = 0
+
+    We use a second-order (piecewise linear) unsplit Godunov method
+    (following Colella 1990).
+
+    Our convection is that the fluxes are going to be defined on the
+    left edge of the computational zones::
+
+        |             |             |             |
+        |             |             |             |
+       -+------+------+------+------+------+------+--
+        |     i-1     |      i      |     i+1     |
+
+                 a_l,i  a_r,i   a_l,i+1
+
+
+    a_r,i and a_l,i+1 are computed using the information in
+    zone i,j.
+
+    Parameters
+    ----------
+    my_data : CellCenterData2d object
+        The data object containing the grid and advective scalar that
+        we are advecting.
+    rp : RuntimeParameters object
+        The runtime parameters for the simulation
+    dt : float
+        The timestep we are advancing through.
+    scalar_name : str
+        The name of the variable contained in my_data that we are
+        advecting
+
+    Returns
+    -------
+    out : ndarray, ndarray
+        The fluxes on the x- and y-interfaces
+
+    """
+
+    myg = my_data.grid
+
+    a = my_data.get_var(scalar_name)
+
+    u = my_data.get_var("x-velocity")
+    v = my_data.get_var("y-velocity")
+
+    cx = myg.scratch_array()
+    cy = myg.scratch_array()
+
+    cx.v(buf=1)[:, :] = u.v(buf=1)*dt/myg.dx
+    cy.v(buf=1)[:, :] = v.v(buf=1)*dt/myg.dy
+
+    # --------------------------------------------------------------------------
+    # monotonized central differences
+    # --------------------------------------------------------------------------
+
+    limiter = rp.get_param("advection.limiter")
+
+    # Give da/dx and da/dy using input: (state, grid, direction, limiter)
+
+    ldelta_ax = reconstruction.limit(a, myg, 1, limiter)
+    ldelta_ay = reconstruction.limit(a, myg, 2, limiter)
+
+    ul_x = myg.scratch_array()
+    ur_x = myg.scratch_array()
+
+    ul_y = myg.scratch_array()
+    ur_y = myg.scratch_array()
+
+    # Determine left and right interface states in x and y.
+
+    # First compute the predictor terms
+
+    ul_x.v(buf=1)[:, :] = a.ip(-1, buf=1) + 0.5*(1.0 - cx.ip(-1, buf=1))*ldelta_ax.ip(-1, buf=1)
+    ul_y.v(buf=1)[:, :] = a.jp(-1, buf=1) + 0.5*(1.0 - cy.jp(-1, buf=1))*ldelta_ay.jp(-1, buf=1)
+    ur_x.v(buf=1)[:, :] = a.v(buf=1) - 0.5*(1.0 + cx.v(buf=1))*ldelta_ax.v(buf=1)
+    ur_y.v(buf=1)[:, :] = a.v(buf=1) - 0.5*(1.0 + cy.v(buf=1))*ldelta_ay.v(buf=1)
+
+    # Solve Riemann's problem to get the correct transverse term
+
+    if scalar_name == "x-velocity":        
+
+        u_xt = riemann(my_data, ul_x, ur_x)
```

Comment:
`direction` defaults to `None` in `riemann`, so you always need to pass in the direction to `riemann()`

Reply:
I was not certain how to proceed with riemann problem. Say that we're advecting x-velocity, `u`, then the shock speed in the x-direction is simply `0.5*(u_l + u_r)`, where `u_l` and `u_r` are the left and right interface states for `u`, which means I don't need to pass in a direction (at least for the way that I wrote the `riemann(), although it could be confusing). But the shock speed in the y-direction we should calculate(?)  that as `(f(u_r) - f(u_l)) / (u_r - u_l)` and here  `f(u_r)` should be `v_r*u_r`, where `v_r` is the right interface velocity in the y-direction, and here is where I use the `direction` option to distinguish whether I switch to calculating the shock speed that way.

Reply:
okay I see what you are saying.  I will try something later.